### PR TITLE
Fix S3253/S6605/SA1625/S103 warnings

### PR DESCRIPTION
**Pull request by iamdmitrij**
# Pull Request

## The issue or feature being addressed
#1290

## S103
- [x] Add `SonarLint.xml` configuration file
- [x] Set `S103` rule's `maximumLineLength` to 500
- [x] Fix inappropriate string lengths > 500 
## Others

- [x] Fix [SA1625](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1625.md)
- [x] Remove [S3253](https://sonarsource.github.io/rspec/#/rspec/S3253/csharp) - no warnings found
- [x] [S6605](https://sonarsource.github.io/rspec/#/rspec/S6605/csharp)

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: src/Polly/Bulkhead/BulkheadSyntax.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public static BulkheadPolicy Bulkhead(int maxParallelization, int maxQueuingActi
     /// <param name="onBulkheadRejected">An action to call, if the bulkhead rejects execution due to oversubscription.</param>
     /// <returns>The policy instance.</returns>
     /// <exception cref="ArgumentOutOfRangeException">maxParallelization;Value must be greater than zero.</exception>
-    /// <exception cref="ArgumentOutOfRangeException">maxParallelization;Value must be greater than zero.</exception>
+    /// <exception cref="ArgumentOutOfRangeException">maxParallelization;Value must be greater than or equal to zero.</exception>
```
Comment (martincostello) [inline src/Polly/Bulkhead/BulkheadSyntax.cs:55]: This is wrong, and without the edit is a duplicate of the line above. Did you mean for it to apply to `maxQueuingActions`?

File: src/Polly/Bulkhead/AsyncBulkheadSyntax.cs [hunk 2]
```diff
@@ -24,7 +24,6 @@ public static AsyncBulkheadPolicy BulkheadAsync(int maxParallelization)
     /// <returns>The policy instance.</returns>
     /// <exception cref="ArgumentOutOfRangeException">maxParallelization;Value must be greater than zero.</exception>
     /// <exception cref="ArgumentNullException">Thrown when <paramref name="onBulkheadRejectedAsync"/> is <see langword="null"/>.</exception>
-    /// <returns>The policy instance.</returns>
```
[the remark this answers is not in the document]
Comment (iamdmitrij) [reply src/Polly/Bulkhead/AsyncBulkheadSyntax.cs:27]: This method had two `returns` definitions on line 24 and line 27, hence one of it was removed to comply with `SA1625` rule.

File: src/Polly/Bulkhead/BulkheadSyntax.cs (hunk 1 above)
Comment (iamdmitrij) [reply src/Polly/Bulkhead/BulkheadSyntax.cs]: Yep, changed reference to `maxQueuingActions`.

File: src/Polly/Bulkhead/AsyncBulkheadSyntax.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (martincostello) [reply src/Polly/Bulkhead/AsyncBulkheadSyntax.cs:27]: Ah, I didn't spot that.

**Pull request merged**